aiosmtpd/tests/test_smtp.py

```diff
@@ -1386,6 +1397,20 @@ def test_exception_handler_exception(self):
                          b'Error: (ValueError) ErroringErrorHandler test')
         self.assertIsInstance(handler.error, ValueError)
 
+    def test_exception_handler_connection_lost_exception(self):
+        handler = ErroringHandlerConnectionLost()
+        controller = Controller(handler)
+        controller.start()
+        self.addCleanup(controller.stop)
+        with SMTP(controller.hostname, controller.port) as client:
+            code, response = client.ehlo('example.com')
+            self.assertEqual(code, 250)
+            with self.assertRaises(SMTPServerDisconnected) as cm:
+                mail = CRLF.join(['Test', '.', 'mail'])
+                client.sendmail('anne@example.com', ['bart@example.com'], mail)
+            self.assertIsInstance(cm.exception, SMTPServerDisconnected)
+            self.assertEqual(handler.error, None)
```

Comment:
Good test. I have some suggestions:

1. After connection is disconnected, ensure that further attempts with `client` raises `SMTPServerDisconnected`. My suggestion:

```python
            # At this point connection should be down
            with self.assertRaises(SMTPServerDisconnected) as cm:
                client.mail("alice@example.com")
            self.assertEqual("please run connect() first", str(cm.exception))
```

2. Add another test case ensuring that existing connections are not impacted. You can have two `SMTP` objects, both already doing their HELO/EHLOs, then first object does a `.sendmail`, and after catching the exception, the second object does a `.mail` (that is, `MAIL FROM`) and this one should get a `250 OK`.

Reply:
I've combined the two suggestions together.
see the:
test_exception_handler_multiple_connections_lost